### PR TITLE
Add flamegraph to collapsed conversion in test framework

### PR DESCRIPTION
**Pull request by roy-soumadipta**
### Description
The commit includes the following changes:

* Addition of flamegraph to collapsed conversion for test assertions.
* Uncommenting flamegraph test assertions.

### Related issues
N/A

### Motivation and context
Flamegraph generation has changed leading to flamegraph test assertion failures. The test framework required changes to accommodate the flamegraph changes.

### How has this been tested?
Tested on AL2_x86_64, aarch64 and macOs

---

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license].

[Apache 2.0 license]: https://www.apache.org/licenses/LICENSE-2.0

### CODE REVIEW

File: test/one/profiler/test/convert/OutputConverter.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import one.convert.Arguments;
+
+import java.io.IOException;
+
+/*
+Interface to be implemented by various profile output converters in tests.
+ */
+public interface OutputConverter<I, O> {
+
+    O convert(I input, Arguments args) throws IOException;
```
Comment (apangin) [inline test/one/profiler/test/convert/OutputConverter.java:17]: 3 new classes added to support one conversion function. Let's not overcomplicate it with an architecture for arbitrary conversions: most likely, there will be just two conversions: flamegraph->collapsed and jfr->collapsed. I'd just add two corresponding methods in `Output` class.

File: test/one/profiler/test/Output.java [hunk 2]
```diff
@@ -51,6 +60,19 @@ public double ratio(String regex) {
         return (double) matched / total;
     }
 
+    public String convert(ProfileOutputType fromOutputType, ProfileOutputType toOutputType, InputStream inputStream) {
```
Comment (apangin) [inline test/one/profiler/test/Output.java:63]: Why accepting `InputStream` and producing `String`?
The source for the conversion is already available in this class as `lines` field. The destination of the conversion is another `Output` instance. So, we convert this `Output` to a new `Output`.

File: test/one/profiler/test/Output.java [hunk 3]
```diff
@@ -51,6 +60,19 @@ public double ratio(String regex) {
...
+        try {
+            if (fromOutputType.equals(FLAMEGRAPH) && toOutputType.equals(COLLAPSED)) {
+                return new FlameToCollapsedConverter().convert(
+                        inputStream, new Arguments("-o", "collapsed"));
+            } else {
+                throw new RuntimeException("Conversion for the provided fromOutputType is not implemented!");
+            }
+        } catch (IOException e) {
+            return "";
```
Comment (apangin) [inline test/one/profiler/test/Output.java:72]: Exceptions should be propagated for the test to fail with the original exception.

File: test/test/kernel/KernelTests.java [hunk 4]
```diff
@@ -19,15 +25,13 @@ public void kernel(TestProcess p) throws Exception {
...
-        }
+        assert err.contains("Kernel symbols are unavailable") || out.contains("sys_getdents");
 
         out = p.profile("stop -o flamegraph");
-//        assert out.contains("f\\(\\d+,\\d+,\\d+,\\d,'java/io/File.list'(,\\d+,\\d+,\\d+)?\\)");
-//        if (!err.contains("Kernel symbols are unavailable")) {
-//            assert out.contains("sys_getdents");
-//        }
+        String convertedOut = out.convert(
+                FLAMEGRAPH, COLLAPSED, new ByteArrayInputStream(out.toString().getBytes(StandardCharsets.UTF_8)));
```
Comment (apangin) [inline test/test/kernel/KernelTests.java:32]: This non-trivial line is repeated in at least two places. A good indicator to modify the API.

File: test/one/profiler/test/Output.java [hunk 5]
```diff
@@ -51,6 +60,19 @@ public double ratio(String regex) {
         return (double) matched / total;
     }
 
+    public String convert(ProfileOutputType fromOutputType, ProfileOutputType toOutputType, InputStream inputStream) {
+        try {
+            if (fromOutputType.equals(FLAMEGRAPH) && toOutputType.equals(COLLAPSED)) {
```
[the remark this answers is not in the document]
Comment (roy-soumadipta) [reply test/one/profiler/test/Output.java]: Enum removed with conversion simplification.

File: test/test/kernel/KernelTests.java (hunk 4 above)
Comment (roy-soumadipta) [reply test/test/kernel/KernelTests.java]: Resolved as part of simplification.

File: test/one/profiler/test/Output.java (hunk 3 above)
Comment (roy-soumadipta) [reply test/one/profiler/test/Output.java]: Rectified

File: test/one/profiler/test/convert/OutputConverter.java (hunk 1 above)
Comment (roy-soumadipta) [reply test/one/profiler/test/convert/OutputConverter.java]: Created thinking of future unknown conversions. Simplified as you mentioned of conversion of profile output being limited to 2 scenarios.

File: test/one/profiler/test/Output.java [hunk 6]
```diff
@@ -51,6 +60,20 @@ public double ratio(String regex) {
         return (double) matched / total;
     }
 
+    public Output convertFlameToCollapsed() throws IOException {
+        FlameGraph fg = new FlameGraph(new Arguments("-o", "collapsed"));
+        try (InputStreamReader in = new InputStreamReader(
+                new ByteArrayInputStream(toString().getBytes(StandardCharsets.UTF_8)), StandardCharsets.UTF_8)) {
```
Comment (apangin) [inline test/one/profiler/test/Output.java:66]: This double UTF-8 conversion String->bytes then bytes->String bothers me a bit.
Perhaps, it's better to use `StringReader` instead.

File: test/one/profiler/test/Output.java [hunk 7]
```diff
@@ -51,6 +60,20 @@ public double ratio(String regex) {
...
+            fg.parseHtml(in);
+        }
+
+        try (ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+             PrintStream out = new PrintStream(outputStream)) {
+            fg.dump(out);
+            return new Output(outputStream.toString().split(System.lineSeparator()));
```
[the remark this answers is not in the document]
Comment (roy-soumadipta) [reply test/one/profiler/test/Output.java]: Addressed

File: test/one/profiler/test/Output.java (hunk 6 above)
Comment (roy-soumadipta) [reply test/one/profiler/test/Output.java]: Addressed

**Pull request merged**